From 427db8b4e36aa6ed9056d600d95f3b113574769b Mon Sep 17 00:00:00 2001
From: Jon Phillips <jphillips@cloudflare.com>
Date: Tue, 30 Apr 2024 00:06:33 +0100
Subject: Expose v8::Symbol::GetDispose().

V8 appears to be in the process of supporting the explicit resource
management spec. Since `Symbol.dispose` has been exposed to JS we need
to be able to access the native v8::Symbol too.

diff --git a/include/v8-primitive.h b/include/v8-primitive.h
index a5d149b9c2884ed79e4185a834f6ccf1b2be9322..f3cfd44df693ff48b5acae20f792357defb7c805 100644
--- a/include/v8-primitive.h
+++ b/include/v8-primitive.h
@@ -785,6 +785,8 @@ class V8_EXPORT Symbol : public Name {
   static Local<Symbol> GetToPrimitive(Isolate* isolate);
   static Local<Symbol> GetToStringTag(Isolate* isolate);
   static Local<Symbol> GetUnscopables(Isolate* isolate);
+  static Local<Symbol> GetDispose(Isolate* isolate);
+  static Local<Symbol> GetAsyncDispose(Isolate* isolate);
 
   V8_INLINE static Symbol* Cast(Data* data) {
 #ifdef V8_ENABLE_CHECKS
diff --git a/src/api/api.cc b/src/api/api.cc
index 582c5dbb81931356b7f22e6c3c562439476264d4..3fba34ba576a779e7c6b2c3d37373e96d906b078 100644
--- a/src/api/api.cc
+++ b/src/api/api.cc
@@ -9654,7 +9654,9 @@ Local<Symbol> v8::Symbol::ForApi(Isolate* v8_isolate, Local<String> name) {
   V(Split, split)                             \
   V(ToPrimitive, to_primitive)                \
   V(ToStringTag, to_string_tag)               \
-  V(Unscopables, unscopables)
+  V(Unscopables, unscopables)                 \
+  V(Dispose, dispose)                         \
+  V(AsyncDispose, async_dispose)              \
 
 #define SYMBOL_GETTER(Name, name)                                      \
   Local<Symbol> v8::Symbol::Get##Name(Isolate* v8_isolate) {           \
